ycmd/http_client.py

```diff
@@ -169,3 +203,17 @@ def CppSemanticCompletionResults(server, path, row, col, contents, filetype='cpp
                                             line_num=row,
                                             column_num=col,
                                             contents=contents)
+
+
+def DefaultSettings(ycmd_path):
+    default_options_path = os.path.join(ycmd_path, "default_settings.json")
```

Comment:
Here I mean that user should have an opportunity to set path to his default_settings.json in plugins settings. This is the last comment - other changes look good to me =)